clippy_lints/src/implicit_saturating_sub.rs

```diff
@@ -16,17 +16,15 @@ declare_clippy_lint! {
     ///
     /// ### Example
     /// ```rust
-    /// let end: u32 = 10;
-    /// let start: u32 = 5;
-    ///
-    /// let mut i: u32 = end - start;
-    ///
-    /// // Bad
```

Comment:
Based on the suggestion and lint description, I'd say, that the subtraction part is still important. Based on the previous example, we should also check if the unsigned type is important :upside_down_face:

Reply:
How about this:

```rust
# let end: u32 = 10;
# let start: u32 = 5;
let mut i: u32 = end - start;

if i != 0 {
    i -= 1;
}
```

I just didn't think `end` and `start` were too important to understanding what the lint does.